src/python_testing/TC_CADMIN_1_15.py

```diff
@@ -0,0 +1,244 @@
+# === END CI TEST ARGUMENTS ===
+
+import logging
+import random
+
+import chip.clusters as Clusters
+from chip import ChipDeviceCtrl
+from chip.ChipDeviceCtrl import CommissioningParameters
+from chip.exceptions import ChipStackError
+from matter_testing_support import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from mobly import asserts
+
+
+class TC_CADMIN_1_15(MatterBaseTest):
+    async def OpenCommissioningWindow(self, th, expectedErrCode) -> CommissioningParameters:
+        if expectedErrCode == 0x00:
+            params = await th.OpenCommissioningWindow(
+                nodeid=self.dut_node_id, timeout=self.max_window_duration, iteration=10000, discriminator=self.discriminator, option=1)
+            return params
+
+        else:
+            ctx = asserts.assert_raises(ChipStackError)
+            with ctx:
+                await th.OpenCommissioningWindow(
+                    nodeid=self.dut_node_id, timeout=self.max_window_duration, iteration=10000, discriminator=self.discriminator, option=1)
+            errcode = ctx.exception.chip_error
+            logging.info('Commissioning complete done. Successful? {}, errorcode = {}'.format(errcode.is_success, errcode))
+            asserts.assert_false(errcode.is_success, 'Commissioning complete did not error as expected')
+            asserts.assert_true(errcode.sdk_code == expectedErrCode,
+                                'Unexpected error code returned from CommissioningComplete')
+
+    async def read_currentfabricindex(self, th: ChipDeviceCtrl) -> int:
+        cluster = Clusters.Objects.OperationalCredentials
+        attribute = Clusters.OperationalCredentials.Attributes.CurrentFabricIndex
+        current_fabric_index = await self.read_single_attribute_check_success(dev_ctrl=th, endpoint=0, cluster=cluster, attribute=attribute)
+        return current_fabric_index
+
+    async def get_fabrics(self, th: ChipDeviceCtrl) -> int:
+        OC_cluster = Clusters.OperationalCredentials
+        fabrics = await self.read_single_attribute_check_success(dev_ctrl=th, fabric_filtered=False, endpoint=0, cluster=OC_cluster, attribute=OC_cluster.Attributes.Fabrics)
+        return fabrics
+
+    async def CommissionAttempt(
+            self, setupPinCode: int, thnum: int, th):
+
+        logging.info(f"-----------------Commissioning with TH_CR{str(thnum)}-------------------------")
+        await th.CommissionOnNetwork(
+            nodeId=self.dut_node_id, setupPinCode=setupPinCode,
+            filterType=ChipDeviceCtrl.DiscoveryFilterType.LONG_DISCRIMINATOR, filter=self.discriminator)
+
+    def generate_unique_random_value(self, value):
+        while True:
+            random_value = random.randint(10000000, 99999999)
+            if random_value != value:
+                return random_value
+
+    def steps_TC_CADMIN_1_15(self) -> list[TestStep]:
+        return [
+            TestStep(1, "Commissioning, already done", is_commissioning=True),
```

Comment:
something's wacky with the step numbering here - doesn't quite match the test plan, i don't think. Would you mind double checking?

Reply:
https://github.com/CHIP-Specifications/chip-test-plans/pull/4636/files

Reply:
the test steps in the test script should now match the test plan